bundler/lib/bundler/cli.rb

```diff
@@ -570,7 +570,7 @@ def viz
     method_option :git, :type => :boolean, :default => true, :desc => "Initialize a git repo inside your library."
     method_option :mit, :type => :boolean, :desc => "Generate an MIT license file. Set a default with `bundle config set gem.mit true`."
     method_option :rubocop, :type => :boolean, :desc => "Add rubocop to the generated Rakefile and gemspec. Set a default with `bundle config set gem.rubocop true`."
-    method_option :test, :type => :string, :lazy_default => "rspec", :aliases => "-t", :banner => "rspec",
```

Comment:
I think we still want to default to Rspec. But if it's a test framework configured use that instead

Reply:
Defaulting to RSpec makes passing empty -t ambiguous, this is the main problem encountered in the original issue. A test framework was configured, but ignored with `lazy_default` defaulting to `rspec`.

If `lazy_default` is set to any test framework, then which of `options[:test]` and `Bundler.settings["gem.test"]` should take precedence?

Reply:
`Bundler.settings["gem.test"]` if it is true else fallback to the default `options[:test]`

With this implementation if you have configured something in `Bundler.settings["gem.test"]` and you pass `-t` when creating a gem no tests are generated.

Reply:
Thanks for catching that error with the current implementation!

Reply:
I incorporated your suggestion of lazy defaulting to `Bundler.settings["gem.test"]`. In addition, if this value is falsy then fallback to `"`. `"` is used to check if we want to reconfirm setting the test framework.